Support for emphasis extensions

thoemmi opened the pull request:
This PR adds support for emphasis extensions (subscript, superscript, strike through, inserted, and marked)
 to both WPF and XAML rendering. All of them can be customized with styles.

This also fixes part of @amkuchta in #10.

thoemmi committed:
added support for emphasis extensions (subscript, superscript, strike through, inserted, and marked)

amkuchta commented:
@thoemmi is there anything you can't do? 😆 😆 😆

thoemmi commented:
@amkuchta  I'm only [sharpening my saw](https://blog.codinghorror.com/sharpening-the-saw/) :wink:

thoemmi committed:
Updated readme (auto-links is already available for XAML renderer as far as I can see)

Kryptos-FR commented:
Thanks. I'll have a look later this week (sorry, quite busy at the moment).

Kryptos-FR commented on src/Markdig.Wpf/MarkdownExtensions.cs:
```diff
@@ -25,7 +25,8 @@ public static MarkdownPipelineBuilder UseSupportedExtensions([NotNull] this Mark
                 .UsePipeTables()
-                .UseTaskLists();
+                .UseTaskLists()
+                .UseEmphasisExtras();
```
I don't remember if the order matters. I think it does sometimes, so maybe  it would be better to follow the same order than https://github.com/lunet-io/markdig/blob/master/src/Markdig/MarkdownExtensions.cs#L59
Note that the order is already different so not a big deal at the moment.

Kryptos-FR commented on src/Markdig.Wpf/Renderers/Xaml/Inlines/EmphasisInlineRenderer.cs:
```diff
@@ -36,7 +36,25 @@ public EmphasisInlineRenderer()
+            renderer.Write("<").Write(tag);
+            switch (obj.DelimiterChar)
+            {
+                case '~':
```
Just for the sake of completeness, add the cases for `'*'` and `'~'` that immediately break. It keeps balance with the previous method and shows that those two options are valid.

Kryptos-FR commented on src/Markdig.Wpf/Styles.cs:
```diff
@@ -86,5 +86,30 @@ public static class Styles
         /// Resource Key for the ThematicBreakStyle.
         /// </summary>
         public static ComponentResourceKey ThematicBreakStyleKey { get; } = new ComponentResourceKey(typeof(Styles), nameof(ThematicBreakStyleKey));
+
```
At some point I should extract the ones that come with an extension into a separate class (maybe one class per extension). For the moment I prefer to have those sorted in alphabetical order.

thoemmi replied:
Sorry, I didn't notice the alphabetic ordering of style keys 😆

Kryptos-FR commented on src/Markdig.Wpf/Styles.cs:
```diff
@@ -86,5 +86,30 @@ public static class Styles
+        public static ComponentResourceKey SubscriptStyleKey { get; } = new ComponentResourceKey(typeof(Styles), nameof(SubscriptStyleKey));
+
+        /// <summary>
+        /// Resource Key for the SubscriptStyle.
```
Copy/paste mistake

Kryptos-FR commented on src/Markdig.Wpf/Styles.cs:
```diff
@@ -86,5 +86,30 @@ public static class Styles
+        public static ComponentResourceKey StrikeThroughStyleKey { get; } = new ComponentResourceKey(typeof(Styles), nameof(StrikeThroughStyleKey));
+
+        /// <summary>
+        /// Resource Key for the InsertedStyleKeyStyle.
```
Copy/paste mistake

Kryptos-FR commented on src/Markdig.Wpf/Styles.cs:
```diff
@@ -86,5 +86,30 @@ public static class Styles
+        public static ComponentResourceKey InsertedStyleKey { get; } = new ComponentResourceKey(typeof(Styles), nameof(InsertedStyleKey));
+
+        /// <summary>
+        /// Resource Key for the MarkedStyleKeyStyle.
```
Copy/paste mistake

Kryptos-FR reviewed (changes requested):
Just a few remarks, nothing critical. I'll merge once the requested changes are done.

Thanks again for the contribution.

thoemmi committed:
sorted style keys alphabetically
explicit cases for bold/italic in EmphasisInlineRenderer for XAML
ordered extensions like original Markdig

thoemmi commented:
I think I've addressed all of your concerns.
